### PR TITLE
Generalize date range entities

### PR DESCRIPTION
**Pull request by kcinay055679**

### CODE REVIEW

File: spec/controllers/advanced_trainings_controller_spec.rb [hunk 1]
```diff
@@ -37,7 +37,8 @@
                                 description: description
                                 }
                               }
-      expect(response).to redirect_to(person_path(person))
+      # TODO: Ask robin
```
Comment (kcinay055679) [inline spec/controllers/advanced_trainings_controller_spec.rb:40]: Ask Robin for the best way to fix

File: app/helpers/date_helper.rb [hunk 2]
```diff
@@ -27,4 +27,12 @@ def months_with_nil
   def last_100_years
     (100.years.ago.year..Time.zone.today.year).to_a.reverse
   end
+
+  def name_of_obj(entry)
+    entry.class.to_s.split('::').first.underscore
```
Comment (kcinay055679) [inline app/helpers/date_helper.rb:32]: Extract those 2 methods to seperate helper

**Review by kcinay055679 — COMMENTED** [review]
Fix notes from review and make sure that the "render_new_after_save" functionallity is working. This will fix the tests

**Comment by kcinay055679** [comment]
Generify the files new and edit from the AdvancedTrainings so every DateRangeEntity can use them

**Review by Robin481 — CHANGES_REQUESTED** [review]
Next time take some time to look through your changes and figure out if you can simplify your code please :rock:

File: app/controllers/advanced_trainings_controller.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
 
-class AdvancedTrainingsController < PersonRelationsController
+class AdvancedTrainingsController < People::PersonRelationsController
   self.permitted_attrs = %i[description year_to month_to year_from month_from person_id]
+  self.nesting = Person
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/controllers/advanced_trainings_controller.rb:5]: no
https://github.com/codez/dry_crud?tab=readme-ov-file#label-Nested+Resources

File: app/helpers/date_helper.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/helpers/date_helper.rb:32]: i extraced this to an other helper

File: app/controllers/dry_crud/generic_model.rb [hunk 4]
```diff
@@ -34,10 +34,14 @@ def path_args(last)
...
+      name = model_ivar_name(plural: plural)
+      instance_variable_get(name) if instance_variable_defined?(name)
+    end
+
+    def model_ivar_name(plural: false)
       name = ivar_name(model_class)
       name = name.pluralize if plural
-      name = :"@#{name}"
-      instance_variable_get(name) if instance_variable_defined?(name)
+      :"@#{name}"
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/controllers/dry_crud/generic_model.rb:44]: Rubocop recommended this syntax

File: app/controllers/crud_controller.rb [hunk 5]
```diff
@@ -84,8 +84,7 @@ def create(**options, &block)
   #
   # Specify a :location option if you wish to do a custom redirect.
 
-  # rubocop:disable Metrics/MethodLength
-  def update(**options, &block)
+  def update(**options, &block) # rubocop:disable Metrics/MethodLength
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/controllers/crud_controller.rb:87]: After reinsering the deleted line rubocop tells me this isn't needed and refers to the [documentation](https://docs.rubocop.org/rubocop/cops_lint.html#lintredundantcopenabledirective)

File: app/helpers/generify_helper.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module GenerifyHelper
+  def name_of_obj(entry)
+    entry.class.to_s.split('::').first.underscore
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/helpers/generify_helper.rb]: changed the name to obj because i use both single items and array with this funktion

**Review by Robin481 — APPROVED** [review]
:rock: :rocket:

**Pull request merged**